app.ts

```diff
@@ -0,0 +1,205 @@
+﻿// Onboarding Project - 2018
+
+let RECORD_COUNT: number; // total number of records available in webserver
+let resizeTimer; // delay required to reduce the number of calls made by browser when resizing the window
+let CONTROLS_HEIGHT;
+let CONTENT_HEIGHT;
+let columnsString; // variable to store columns text from api call
+let recordsText; // variable to store records from api call
+let columnsArray;
+
+// makes api call and returns data
+function getRecordCount() {
+	const apiCall = new XMLHttpRequest();
+
+	apiCall.onreadystatechange = function () {
+		if (apiCall.readyState === 4 && apiCall.status === 200) {
+			RECORD_COUNT = JSON.parse(apiCall.response);
+
+			if (columnsString !== undefined) {
+				tableElement.generateTable();
+			}
+		}
+	}
+	apiCall.open("GET", "http://localhost:2050/recordCount", true);
+	apiCall.send();
+}
+
+function getCols() {
+	const apiCall = new XMLHttpRequest();
+	apiCall.onreadystatechange = function () {
+		if (apiCall.readyState === 4 && apiCall.status === 200) {
+			columnsString = apiCall.response;
+			columnsArray = parseText(columnsString);
+			if (RECORD_COUNT !== undefined) {
+				tableElement.generateTable();
+			}
+		}
+	}
+	apiCall.open("GET", "http://localhost:2050/columns", true);
+	apiCall.send();
+}
+
+function getTable(start: number, end: number, tableElement) {
+	const apiCall = new XMLHttpRequest();
+
+	const loadDiv = document.createElement('div');
+	loadDiv.setAttribute('class', 'loader')
+	tableElement.myTable.parentNode.appendChild(loadDiv);
+
+	apiCall.onreadystatechange = function () {
```

Comment:
1. rather use () => { }
2. even better, make this a separate function with a name that describes what it is does.
3. can you not define this function and the parent `getTable` as methods instead?

Reply:
suggestions don't seem to work

Reply:
thank you